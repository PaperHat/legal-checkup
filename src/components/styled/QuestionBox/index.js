import styled from 'styled-components'

const QuestionBox = styled.div`
  width: 260px;
  height: 120px;
  font-family: Helvetica;
  font-size: 25px;
  font-weight: normal;
  font-style: normal;
  font-stretch: normal;
  line-height: normal;
  letter-spacing: normal;
  text-align: center;
`
QuestionBox.displayName = 'StyledQuestionBox'

export default QuestionBox
