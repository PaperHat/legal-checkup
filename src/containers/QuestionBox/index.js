import StyledQuestionBox from '@components/styled/QuestionBox'
import { connect } from 'react-redux'
import { getActiveQuestion } from '@state/selectors'

function mapStateToProps (state) {
  const { body: children } = getActiveQuestion(state)

  return { children }
}

export default connect(mapStateToProps)(StyledQuestionBox)
