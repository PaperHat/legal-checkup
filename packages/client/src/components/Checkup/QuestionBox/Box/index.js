import styled from 'styled-components'

const Box = styled.div`
  width: 260px;
  height: 120px;
  font-family: Helvetica;
  font-size: 25px;
  font-weight: normal;
  font-style: normal;
  font-stretch: normal;
  line-height: normal;
  letter-spacing: normal;
  text-align: center;
`
Box.displayName = 'Box'

export default Box
