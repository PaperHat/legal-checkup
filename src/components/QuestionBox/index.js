import React from 'react'
import { StyledQuestionBox } from '../styled'

export default function QuestionBox ({ questions, activeQuestion }) {
  return <StyledQuestionBox>{questions[activeQuestion].body}</StyledQuestionBox>
}
/* check where to pass question box component to be rendered
  it goes through the home page or checkup page?
  */
