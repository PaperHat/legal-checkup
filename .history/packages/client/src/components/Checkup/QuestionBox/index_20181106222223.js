import { connect } from 'react-redux'

import { getActiveQuestion } from '../../../state/selectors'

import Box from './Box'

function mapStateToProps (state) {
  const { body: children } = getActiveQuestion(state)

  return { children }
}

export default connect(mapStateToProps)(Box)
