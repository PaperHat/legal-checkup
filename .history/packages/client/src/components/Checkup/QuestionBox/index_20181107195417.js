import { connect } from 'react-redux'

import { getActiveQuestion } from '../../../state/selectors'

import Box from './Box'

function mapStateToProps ({ checkup }) {
  const { body: children } = getActiveQuestion(checkup)

  return { children }
}

export default connect(mapStateToProps)(Box)
