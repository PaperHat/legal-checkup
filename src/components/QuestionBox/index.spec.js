import QuestionBox from '.'
import React from 'react'
import { shallow } from 'enzyme'

describe('components:QuestionBox', () => {
  it(`renders the QuestionBox component when a question is provided`, () => {
    const body = 'Is anyone chasing you for money?'

    expect(shallow(<QuestionBox body={body} />)).toMatchSnapshot()
  })

  it(`returns null when a question is not provided`, () => {
    expect(shallow(<QuestionBox />)).toBeNull()
  })
})
