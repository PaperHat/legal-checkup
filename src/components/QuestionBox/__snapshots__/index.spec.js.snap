// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components:QuestionBox renders the QuestionBox and CSS properties properly 1`] = `
<StyledQuestionBox>
  question?
</StyledQuestionBox>
`;
